### PR TITLE
Remove YAML; add disclosures to SD-JWT examples; add protected headers

### LINKED ISSUE #264: Improving the SD-JWT examples
The current examples of SD-JWT in section 3.2, whilst correct, only comprise the credential with the SD fields flagged (using !sd) prior to securing, followed by the SD-JWT secured verifiable credential. It may be helpful to the reader to show the contents of the unencoded JWT body, viz:
```
{
  "_sd_alg": "sha-256",
  "@context": [
    "https://www.w3.org/ns/credentials/v2",
    "https://www.w3.org/ns/credentials/examples/v2"
  ],
  "issuer": "https://university.example/issuers/565049",
  "validFrom": "2010-01-01T19:23:24Z",
  "credentialSchema": {
    "_sd": [
      "C7P4_rPSwJJ63kDm6K7PI4E39IXCDu_a98tmI-8UlwA",
      "fechAzukKKvWk_c3JfU-Sk_GVcX3CFGeuJfN6MNL_uQ"
    ]
  },
  "credentialSubject": {
    "degree": {
      "name": "Bachelor of Science and Arts",
      "_sd": [
        "wy-UoUQJmXJndaMROqTapQGJ8EUU8Dz8LwtCO0g3gnU"
      ]
    },
    "_sd": [
      "MCvCBoKRoUiivsPEfg00nAuGr8DwaWUeAkHuBNq4zdI"
    ]
  },
  "_sd": [
    "2D5lkIK2hrAd2a4QK_dtoJI8ZEOgKYaXKhOngaxvRsc",
    "WTfYOzSQhejkIqe5IYDkNJrqw5r4KpxHgOaaV04hsMQ"
  ]
}
```
and add some description of the SD-JWT such as:  "the six fields preceded by a tilde that follow the JWT signature are base64 encodings of the six selectively disclosed properties"

### LINKED ISSUE #269: Intellectual Property concern wrt. SD-JWT examples
@bc-pi wrote (on the [VCWG mailing list](https://lists.w3.org/Archives/Public/public-vc-wg/2024May/0003.html):
> Are there any concerns from the W3C or the Verifiable Credentials Working Group regarding the use of some of the YAML constructs from the OpenWallet Foundation’s SD-JWT Reference Implementation in the 'Securing Verifiable Credentials using JOSE and COSE' publication? Specifically, these constructs appear in two examples* without attribution, explanation or reference. Could this raise issues related to publication process/procedure, intellectual property rights, or document clarity?
> \* these two examples:
https://www.w3.org/TR/vc-jose-cose/#example-a-simple-example-of-a-verifiable-credential-secured-with-sd-jwt
https://www.w3.org/TR/vc-jose-cose/#example-a-simple-example-of-a-verifiable-presentation-secured-with-sd-jwt

### LINKED ISSUE #271: Show headers in examples
The examples show compact JWT and CWT text and payloads, but none show the header.  The header is the most interesting part in some ways.

### PR DESCRIPTION
**Pull request by decentralgabe**
Fix #269 and replaces #270
Fix #271

Does not yet address #264 which I will take in a subsequent PR to add multiple tabs to the standalone SD-JWT example

for the combined example I feel this would be too much to display

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/pull/272.html" title="Last updated on May 16, 2024, 4:02 AM UTC (ff4c401)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/272/5ba3d8d...ff4c401.html" title="Last updated on May 16, 2024, 4:02 AM UTC (ff4c401)">Diff</a>

### CODE REVIEW

**Review by selfissued — APPROVED** [review]

File: plugin/src/exampleJwt.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import {holder, issuer, key, text} from "@transmute/verifiable-credentials";
+import {base64url, holder, issuer, key, text} from "@transmute/verifiable-credentials";
```
Comment (TallTed) [inline plugin/src/exampleJwt.js:1]:
```suggestion
import { base64url, holder, issuer, key, text } from "@transmute/verifiable-credentials";
```

File: plugin/src/exampleSdJwt.js [hunk 2]
```diff
@@ -1,38 +1,81 @@
...
-    const edited1 = claims.replace(/  !sd id\:(.*?)\n/g, `  id: False\n`)
-    // disclose types
-    const edited2 = edited1.replace(/\!sd type\:/g, `type:`)
-    // redact remaining ids
-    return edited2.replace(/\!sd id\:/g, `id:`)
-}
-
-const getSdHtml = (vc) =>{
+import yaml from 'yaml';
+import {base64url, issuer, key, text} from "@transmute/verifiable-credentials";
```
Comment (TallTed) [inline plugin/src/exampleSdJwt.js:2]:
```suggestion
import { base64url, issuer, key, text } from "@transmute/verifiable-credentials";
```

**Review by TallTed — CHANGES_REQUESTED** [review]
A little cross-file consistency

File: plugin/src/exampleSdJwt.js [hunk 3]
```diff
@@ -1,38 +1,81 @@
...
+import * as jose from 'jose';
+import crypto from 'crypto'; // Ensure you have this dependency
```
Comment (decentralgabe) [inline plugin/src/exampleSdJwt.js:4]:
```suggestion
import crypto from 'crypto';
```

**Comment by bc-pi** [comment]
Not sure if/how to comment here but from a quick look at the [Preview](https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/pull/272.html) the examples don't look to self consistent or syntactically correct.

**Comment by decentralgabe** [comment]
@bc-pi I see some issues too, I'll fix them and comment here when they're addressed.

**Review by bc-pi — CHANGES_REQUESTED** [review]
https://github.com/w3c/vc-jose-cose/pull/270#issuecomment-2116236301 has some candid thoughts

**Comment by decentralgabe** [comment]
@bc-pi it's rarely I screw up so bad I leave someone speechless 😂  - my apologies

I've fixed the code, if you refresh https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/pull/272.html you'll see it in a better state. I may have to do some work on the nested disclosures, though I still believe this is an improvement.

Hopefully it's at least to a place where you can articulate what's still not right.

**Comment by decentralgabe** [comment]
going to close and re-open ... rendering is not working as expected

**Pull request closed without merging**